pkg/files/file_helpers.go

```diff
@@ -139,3 +139,62 @@ func CompareFileHash(fileOverview *mpi.FileOverview) (fileDiff map[string]*mpi.F
 
 	return fileDiff, fileContents, nil
 }
+
+// DetermineFileAction compares two lists of files, determines which file action is needed for each file,
+// then returns a list of files with the action set for each file
+func DetermineFileAction(currentFiles, modifiedFiles []*mpi.File) []*mpi.File {
+	// Go doesn't allow address of numeric constant
+	addAction := mpi.File_FILE_ACTION_ADD
+	updateAction := mpi.File_FILE_ACTION_UPDATE
+	deleteAction := mpi.File_FILE_ACTION_DELETE
+	unchangedAction := mpi.File_FILE_ACTION_UNCHANGED
+
+	filesWithActions := make([]*mpi.File, 0, len(modifiedFiles))
+	currentFilesMap := convertToMapOfFiles(currentFiles)
+	modifiedFilesMap := convertToMapOfFiles(modifiedFiles)
+
+	// if file is in currentFiles but not in modified files, file has been deleted
+	for _, currentFile := range currentFilesMap {
+		_, ok := modifiedFilesMap[currentFile.GetFileMeta().GetName()]
+		if !ok {
+			currentFile.Action = &deleteAction
+			filesWithActions = append(filesWithActions, currentFile)
+			delete(modifiedFilesMap, currentFile.GetFileMeta().GetName())
```

Comment:
Didn't we check above that `currentFile.GetFileMeta().GetName` is *not* in `modifiedFilesMap`?